cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -244,14 +250,50 @@ public void getStatus(HttpRequest request, HttpResponder responder,
     }
   }
 
+  private void getScheduleStatus(HttpResponder responder, String appId, String namespaceId, String scheduleName) {
+    try {
+      ApplicationSpecification appSpec = store.getApplication(Id.Application.from(namespaceId, appId));
+      if (appSpec == null) {
+        responder.sendString(HttpResponseStatus.NOT_FOUND, "App: " + appId + " not found");
+        return;
+      }
+
+      ScheduleSpecification scheduleSpec = appSpec.getSchedules().get(scheduleName);
+      if (scheduleSpec == null) {
+        responder.sendString(HttpResponseStatus.NOT_FOUND, "Schedule: " + scheduleName + " not found");
+        return;
+      }
+
+      String programName = scheduleSpec.getProgram().getProgramName();
+      Id.Program programId = Id.Program.from(namespaceId, appId, programName);
+      JsonObject json = new JsonObject();
+      json.addProperty("status", scheduler.scheduleState(programId, scheduleSpec.getProgram().getProgramType(),
+                                                         scheduleName).toString());
+      responder.sendJson(HttpResponseStatus.OK, json);
+    } catch (SecurityException e) {
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (Throwable e) {
+      LOG.error("Got exception:", e);
+      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+
+
+
   @POST
   @Path("/apps/{app-id}/{runnable-type}/{runnable-id}/{action}")
-  public void startStopDebugProgram(HttpRequest request, HttpResponder responder,
+  public void takeActionOnProgram(HttpRequest request, HttpResponder responder,
```

Comment:
+1 to `performAction` or `handleAction`.

Also, since now we rely on this method to handle both program action as well as schedule action, better rename the parameter to `type` and `id`, since it may not be runnable (actually it never be runnable, it was only program-type and program-id).